src/main/java/org/osc/manager/ism/api/IsmDomainApi.java

```diff
@@ -18,35 +18,104 @@
     }
 
-    public static IsmDomainApi create(ApplianceManagerConnectorElement mc) throws Exception {
-        return new IsmDomainApi(mc);
+    public String createDomain(ManagerDomainElement dm) throws Exception {
+
+        String name = this.mc.getName();
+        DomainEntity domains = new DomainEntity();
+        domains.setName(dm.getName());
+
+        return this.txControl.required(new Callable<DomainEntity>() {
+            @Override
+            public DomainEntity call() throws Exception {
+                CriteriaBuilder cb = IsmDomainApi.this.em.getCriteriaBuilder();
+
+                CriteriaQuery<ApplianceManagerConnectorEntity> q = cb
+                        .createQuery(ApplianceManagerConnectorEntity.class);
+                Root<ApplianceManagerConnectorEntity> r = q.from(ApplianceManagerConnectorEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("name"), name)));
+                ApplianceManagerConnectorEntity result = IsmDomainApi.this.em.createQuery(q).getSingleResult();
+                domains.setapplianceManagerConnector(result);
+                IsmDomainApi.this.em.persist(domains);
+                return domains;
+            }
+        }).getName();
     }
 
     @Override
-    public Domain getDomain(String domainId) throws Exception {
-        return new Domain(0L, "Root-Domain");
+    public DomainEntity getDomain(String domainId) throws Exception {
+
+        return this.txControl.required(new Callable<DomainEntity>() {
+            @Override
+            public DomainEntity call() throws Exception {
+                CriteriaBuilder cb = IsmDomainApi.this.em.getCriteriaBuilder();
+
+                CriteriaQuery<DomainEntity> q = cb.createQuery(DomainEntity.class);
+                Root<DomainEntity> r = q.from(DomainEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("id"), domainId)));
+                DomainEntity result = IsmDomainApi.this.em.createQuery(q).getSingleResult();
+                return result;
+            }
+        });
     }
 
     @Override
-    public List<DomainListElement> listDomains() throws Exception {
-        List<DomainListElement> domainList = new ArrayList<DomainListElement>();
-        domainList.add(new DomainListElement(0L, "Root-Domain"));
-        return domainList;
+    public List<DomainEntity> listDomains() throws Exception {
+
+        List<DomainEntity> domains = new ArrayList<DomainEntity>();
+
+        return this.txControl.supports(new Callable<List<DomainEntity>>() {
+
+            @Override
+            public List<DomainEntity> call() throws Exception {
+                CriteriaBuilder cb = IsmDomainApi.this.em.getCriteriaBuilder();
+                CriteriaQuery<DomainEntity> q = cb.createQuery(DomainEntity.class);
+                Root<DomainEntity> r = q.from(DomainEntity.class);
+                q.select(r);
+
+                for (DomainEntity domain : IsmDomainApi.this.em.createQuery(q).getResultList()) {
```

Comment:
No I get Exception illegal state exception and after search in web, I understood that rest response is sent before data is populated.
by copying the list element, I don't get the above error.